### DISCUSSION TITLE
The WireGuard protocol saves your IP address on the VPN server for a long period of time

### DISCUSSION THREAD

**Discussion by AdanC1b3r** [pivpn/pivpn#1595] [Q&A]
Does PiVPN mitigate that problem or what solution does it have to this?

**Comment by coolapso** [comment]
why is that a problem?
Once again you are misunderstanding the use-case of pivpn!


converting to Discussion, not an issue

**Reply by AdanC1b3r** [reply]
WireGuard requires that allowed IP addresses be stored on the server until the server restarts.

From a security standpoint, it's concerning, because if the server gets compromised, someone could use the IP address to link it to your activity and therefore the main advantage of using a VPN is overridden.

**Reply by AdanC1b3r** [reply]
Most commercial VPN services supported by WireGuard have implemented solutions to minimize these privacy risks.

For example, delete your IP address if three minutes of inactivity elapse. Also, create a new IP address randomly every 24 hours to avoid the problems of using a static IP address.

**Reply by TinCanTech** [reply]
If you don't like wireguard then use openvpn

**Reply by coolapso** [reply]
you have a point however it is not valid for the typical pivpn use-case. which is far from the commercial VPN use-case. 

On a common user scenario (which is the PiVPN target) you will either have it at home to access your home network or to use public networks securely
Another scenario is that you can actually have it on a cloud Instance/VPS which you pay for with your credit card therefore activity can always be linked to you. 

So if your goal is to really hide your (shady) activity from someone, then you are better off with a commercial VPN and PiVPN does not suit your needs.

**Reply by AdanC1b3r** [reply]
> If you don't like wireguard then use openvpn

Wireguard is spectacular with encryption algorithms and state-of-the-art authentication protocols, much better than OpenVPN, only it has the problem I told you earlier, if the server is compromised you can get your IP, why it stores it on the server until it restarts, that can be mitigated.

**Reply by TinCanTech** [reply]
OpenVPN has **decades** of proven security excellence. I don't know about you ..

**Reply by coolapso** [reply]
Well its a proven fact that most security issues are due to bad configs / practices and not the protocols / software per se!

**Reply by TinCanTech** [reply]
I can count, _on one hand_, the serious security issues that OpenVPN has been guilty of. I even discovered one myself: https://community.openvpn.net/openvpn/ticket/1304 (Fixed in less than one day)

**Reply by AdanC1b3r** [reply]
Interesting, it has had many vulnerabilities, which is why Wireguard promises state-of-the-art encryption.

**Reply by TinCanTech** [reply]
Promises are often broken, much like hearts and Windows.

**Reply by orazioedoardo** [reply]
> From a security standpoint, it's concerning, because if the server gets compromised, someone could use the IP address to link it to your activity and therefore the main advantage of using a VPN is overridden.

PiVPN generates configuration files for the vanilla WireGuard app. Supporting AllowedIPs rotation would require us to create a custom client with an out-of-band mechanism to change IP addresses.

> Wireguard is spectacular with encryption algorithms and state-of-the-art authentication protocols, much better than OpenVPN, only it has the problem I told you earlier

Note that we also use static virtual IPs with OpenVPN to support resolving `clientname.pivpn` to the virtual IP when using Pi-hole.
